yamale/schema/schema.py

```diff
@@ -60,77 +63,97 @@ def validate(self, data, data_name):
                 error_str = error_str.encode('utf-8')
             raise ValueError(error_str)
 
-    def _validate(self, validator, data, path, includes=None):
+    def _validate_item(self, validator, data, path, strict, key):
         """
-        Run through a schema and a data structure,
-        validating along the way.
-
-        Ignores fields that are in the data structure, but not in the schema.
+        Fetch item from data at the postion key and validate with validator.
 
         Returns an array of errors.
         """
         errors = []
-
+        path = path + DataPath(key)
         try:  # Pull value out of data. Data can be a map or a list/sequence
-            data_item = path.getitem(data)
+            data_item = data[key]
         except KeyError:  # Oops, that field didn't exist.
+            # TODO add check if is validator
             if validator.is_optional:  # Optional? Who cares.
                 return errors
             # SHUT DOWN EVERTYHING
             errors.append('%s: Required field missing' % path)
             return errors
 
-        return self._validate_item(validator, data_item, data, path, includes)
+        return self._validate(validator, data_item, path, strict)
 
-    def _validate_item(self, validator, data_item, data, path, includes):
+    def _validate(self, validator, data, path, strict):
```

Comment:
I swapped name of the functions here since I think it makes more sense this way now. Maybe I should also swap place of the functions since validate is the first function called. The diff is nicer this way though..